src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,245 @@
+        if (userInput.contains(" ")) {
+            String description = getDescriptionOfTask(userInput);
+
+            switch (command) {
+            case TODO:
+                tasks.add(new ToDo(description));
+                break;
+            case DEADLINE:
+                if (description.contains("/by")) {
+                    String[] inputs = separateDescriptionAndTime(description);
+                    tasks.add(new Deadline(inputs[0], inputs[1]));
+                } else {
+                    throw new InvalidDescriptionException();
+                }
+                break;
+            case EVENT:
+                if (description.contains("/at")) {
+                    String[] inputs = separateDescriptionAndTime(description);
+                    tasks.add(new Event(inputs[0], inputs[1]));
+                } else {
+                    throw new InvalidDescriptionException();
+                }
+                break;
+            }
+        } else {
+            throw new NoDescriptionException();
+        }
+    }
+
+    public static CommandType scanCommand(String userInput) throws InvalidCommandException {
+        CommandType command;
+        if (userInput.equals("bye")) {
+            command = CommandType.BYE;
+        } else if (userInput.equals("list")) {
+            command = CommandType.LIST;
+        } else if (userInput.startsWith("done")) {
+            command = CommandType.DONE;
+        } else if (userInput.startsWith("delete")) {
+            command = CommandType.DELETE;
+        } else if (userInput.startsWith("todo")) {
+            command = CommandType.TODO;
+        } else if (userInput.startsWith("deadline")) {
+            command = CommandType.DEADLINE;
+        } else if (userInput.startsWith("event")) {
+            command = CommandType.EVENT;
+        } else {
+            throw new InvalidCommandException();
+        }
+        return command;
+    }
+
+    public static void manageTasks() {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        boolean runDuke = true;
+        File f = new File("duke.txt");
+        try {
+            loadPrevData(tasks, "duke.txt");
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        Scanner in = new Scanner(System.in);
+        while (runDuke) {
+            String userInput = in.nextLine();
+            try {
+                CommandType command = scanCommand(userInput);
+                switch (command) {
+                case BYE:
+                    runDuke = false;
+                    break;
+                case LIST:
+                    printList(tasks);
+                    break;
+                case DONE:
+                    int completedTaskNo = Integer.parseInt(userInput.replace("done ", ""));
+                    tasks.get(completedTaskNo - 1).markAsDone();
```

Comment:
This line seems to be doing two things at once, which is [undesirable](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions). You could consider adding another variable to keep count instead.